src/Integration.Vsix.UnitTests/Events/ProjectDocumentsEventsListenerTests.cs

```diff
@@ -0,0 +1,217 @@
+    }
+
+    [TestMethod]
+    public void OnQueryAddFiles_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnQueryAddFiles(null, 1, ["C:\\Users\\test\\TestProject\\AFile.cs"], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnAfterAddDirectoriesEx_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnAfterAddDirectoriesEx(1, 1, [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnAfterRemoveDirectories_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnAfterRemoveDirectories(1, 1, [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnQueryRenameFiles_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnQueryRenameFiles(null, 1, [], [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnQueryRenameDirectories_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnQueryRenameDirectories(null, 1, [], [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnAfterRenameDirectories_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnAfterRenameDirectories(1, 1, [], [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnQueryAddDirectories_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnQueryAddDirectories(null, 1, [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnQueryRemoveFiles_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnQueryRemoveFiles(null, 1, [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnQueryRemoveDirectories_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnQueryRemoveDirectories(null, 1, [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void OnAfterSccStatusChanged_DoesNothing()
+    {
+        var testSubject = CreateTestSubject();
+
+        var result = testSubject.OnAfterSccStatusChanged(1, 1, [], [], [], []);
+
+        result.Should().Be(VSConstants.S_OK);
+    }
+
+    [TestMethod]
+    public void Dispose_UnadviseTrackProjectDocumentsEvents()
+    {
+        var trackProjectDocuments2 = Substitute.For<IVsTrackProjectDocuments2>();
```

Comment:
This is interesting. You're using a mock, but then the test failed with a real exception

```
Test method SonarLint.VisualStudio.Integration.UnitTests.Events.ProjectDocumentsEventsListenerTests.Dispose_UnadviseTrackProjectDocumentsEvents threw exception:
System.Runtime.InteropServices.COMException: Dispose must be called on the UI thread.. If you are using UI objects in test consider using [UITestMethod] attribute instead of [TestMethod] to execute test in UI thread.


```